platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/wiring/PlatformCoordinator.java

```diff
@@ -107,11 +112,12 @@ public void flushIntakePipeline() {
         eventDeduplicatorWiring.flushRunnable().run();
         eventSignatureValidatorWiring.flushRunnable().run();
         orphanBufferWiring.flushRunnable().run();
-        eventCreationManagerWiring.flush();
         inOrderLinkerWiring.flushRunnable().run();
         shadowgraphWiring.flushRunnable().run();
         consensusEngineWiring.flushRunnable().run();
         applicationTransactionPrehandlerWiring.flushRunnable().run();
+        futureEventBufferWiring.flushRunnable().run();
+        eventCreationManagerWiring.flush();
```

Comment:
Not clear why this was re-ordered.  If the order these are called is important, however, it might be good to add a comment to that effect at end-of-line.

Reply:
Order for this is super important! The original ordering was incorrect.

I've added the following comment in the function:

```
        // Important: the order of the lines within this function matters. Do not alter the order of these
        // lines without understanding the implications of doing so. Consult the wiring diagram when deciding
        // whether to change the order of these lines.
```

I also added this to the javadoc, since the contract of what this method is attempting to achieve is not straight forward:

```
    /**
     * Flushes the intake pipeline. After this method is called, all components in the intake pipeline (i.e. components
     * prior to the consensus engine) will have been flushed. Additionally, things will be flushed an order that
     * guarantees that there will be no remaining work in the intake pipeline (as long as there are no additional events
     * added to the intake pipeline, and as long as there are no events released by the orphan buffer).
     */
```

If you'd like to understand better how this part of the platform is wired up, I'd suggest taking a look at the latest iteration of the [wiring diagram](https://mermaid.ink/svg/JSV7aW5pdDogeydmbG93Y2hhcnQnOiB7J2RlZmF1bHRSZW5kZXJlcic6ICdlbGsnfX19JSUKZmxvd2NoYXJ0IFRECnYwWy8iQ29uc2Vuc3VzIEV2ZW50IFN0cmVhbSIvXQpzdWJncmFwaCB2MVsiQ29uc2Vuc3VzIFBpcGVsaW5lIl0KdjJbIkNvbnNlbnN1cyBFbmdpbmUiXQp2M1siaW5PcmRlckxpbmtlcjxiciAvPvCfjIAiXQp2NCgoIvCfjIAiKSkKdjUoKCLwn5OsIikpCnY2KCgi8J+avSIpKQplbmQKc3ViZ3JhcGggdjdbIkV2ZW50IENyZWF0aW9uIl0KdjhbImV2ZW50Q3JlYXRpb25NYW5hZ2VyPGJyIC8+4p2k77iP8J+MgCJdCnY5WyJmdXR1cmVFdmVudEJ1ZmZlcjxiciAvPvCfjIAiXQp2MTB7eyJmdXR1cmVFdmVudEJ1ZmZlclNwbGl0dGVyIn19CnYxMVsvInRyYW5zYWN0aW9uUG9vbCIvXQp2MTIoKCLwn42OIikpCmVuZApzdWJncmFwaCB2MTNbIkV2ZW50IEhhc2hpbmciXQp2MTRbWyJldmVudEhhc2hlciJdXQp2MTVbInBvc3RIYXNoQ29sbGVjdG9yIl0KZW5kCnN1YmdyYXBoIHYxNlsiRXZlbnQgVmFsaWRhdGlvbiJdCnYxN1siZXZlbnREZWR1cGxpY2F0b3I8YnIgLz7wn4yAIl0KdjE4WyJldmVudFNpZ25hdHVyZVZhbGlkYXRvcjxiciAvPvCfjIAiXQp2MTlbImludGVybmFsRXZlbnRWYWxpZGF0b3I8YnIgLz7wn42OIl0KZW5kCnN1YmdyYXBoIHYyMFsiR29zc2lwIl0KdjIxe3siZ29zc2lwIn19CnYyMlsic2hhZG93Z3JhcGg8YnIgLz7wn4yA8J+TrCJdCmVuZApzdWJncmFwaCB2MjNbIkhlYXJ0YmVhdCJdCnYyNFsiaGVhcnRiZWF0Il0KdjI1KCgi4p2k77iPIikpCmVuZAp2MjZbIk9ycGhhbiBCdWZmZXI8YnIgLz7wn4yAIl0Kc3ViZ3JhcGggdjI3WyJQQ0VTIFJlcGxheSJdCnYyOFsvInBjZXNSZXBsYXllciIvXQp2MjkoKCLinIUiKSkKZW5kCnN1YmdyYXBoIHYzMFsiUHJlY29uc2Vuc3VzIEV2ZW50IFN0cmVhbSJdCnYzMVsvImV2ZW50RHVyYWJpbGl0eU5leHVzIi9dCnYzMlsvInBjZXNTZXF1ZW5jZXIiL10KdjMzWyJwY2VzV3JpdGVyPGJyIC8+4pyF8J+MgPCfk4Dwn5q9Il0KdjM0KCgi8J+VkSIpKQplbmQKc3ViZ3JhcGggdjM1WyJTaWduYXR1cmUgTWFuYWdlbWVudCJdCnN1YmdyYXBoIHYzNlsiSXNzIERldGVjdG9yIl0KdjM3e3siZXh0cmFjdFNpZ25hdHVyZXNGb3JJc3NEZXRlY3RvciJ9fQp2MzhbImlzc0RldGVjdG9yIl0KdjM5Wy8iaXNzSGFuZGxlciIvXQp2NDBbLyJpc3NOb3RpZmljYXRpb25FbmdpbmUiL10KdjQxe3siaXNzTm90aWZpY2F0aW9uU3BsaXR0ZXIifX0KdjQyWy8ic3RhdHVzTWFuYWdlcl9zdWJtaXRDYXRhc3Ryb3BoaWNGYWlsdXJlIi9dCmVuZAp2NDNbIlN0YXRlIFNpZ25hdHVyZSBDb2xsZWN0aW9uIl0KdjQ0Wy8ibGF0ZXN0Q29tcGxldGVTdGF0ZU5leHVzIi9dCnY0NVsibGF0ZXN0Q29tcGxldGVTdGF0ZU5vdGlmaWNhdGlvbiJdCnY0Nlsic3RhdGVTaWduZXIiXQplbmQKdjQ3WyJTdGF0ZSBGaWxlIE1hbmFnZW1lbnQiXQpzdWJncmFwaCB2NDhbIlN0YXRlIE1vZGlmaWNhdGlvbiJdCnY0OVsiY29uc2Vuc3VzUm91bmRIYW5kbGVyPGJyIC8+8J+UrvCflZEiXQp2NTB7eyJydW5uaW5nSGFzaFVwZGF0ZSJ9fQplbmQKc3ViZ3JhcGggdjUxWyJUcmFuc2FjdGlvbiBQcmVoYW5kbGluZyJdCnY1MltbImFwcGxpY2F0aW9uVHJhbnNhY3Rpb25QcmVoYW5kbGVyIl1dCnY1MygoIvCflK4iKSkKZW5kCnY1NFsiaGFzaExvZ2dlciJdCnY1NSgoIvCfk4AiKSkKdjIgLS0gInJvdW5kcyIgLS0+IHYwCnYyIC0tICJyb3VuZHMiIC0tPiB2NDkKdjIgLS4gIm5vbi1hbmNpZW50IGV2ZW50IHdpbmRvdyIgLi0+IHY0CnYyIC0tICJmbHVzaCByZXF1ZXN0IiAtLT4gdjYKdjI2IC0tICJwcmVjb25zZW5zdXMgZXZlbnRzIiAtLT4gdjQzCnYyNiAtLSAicHJlY29uc2Vuc3VzIGV2ZW50cyIgLS0+IHY1Mgp2MjYgLS0gInByZWNvbnNlbnN1cyBldmVudHMiIC0tPiB2OQp2MjYgLS0gImV2ZW50cyB0byBzZXF1ZW5jZSIgLS0+IHYzMgp2NDcgLS0gIm1pbmltdW0gaWRlbnRpZmllciB0byBzdG9yZSIgLS0+IHY1NQp2NDMgLS0gInN0YXRlcyIgLS0+IHY0Nwp2NDMgLS0gImNvbXBsZXRlIHN0YXRlcyIgLS0+IHY0NAp2NDMgLS0gImNvbXBsZXRlIHN0YXRlcyIgLS0+IHY0NQp2NTIgLS4gImZ1dHVyZXMiIC4tbyB2NTMKdjggLS4gImdldCB0cmFuc2FjdGlvbnMiIC4tbyB2MTEKdjggLS4gIm5vbi12YWxpZGF0ZWQgZXZlbnRzIiAuLT4gdjEyCnYxNyAtLSAiZXZlbnRzIHdpdGggdW52YWxpZGF0ZWQgc2lnbmF0dXJlcyIgLS0+IHYxOAp2MzEgLS4gIndhaXQgZm9yIGR1cmFiaWxpdHkiIC4tbyB2MzQKdjE0IC0tICJoYXNoZWQgZXZlbnRzIiAtLT4gdjE1CnYxOCAtLSAidW5vcmRlcmVkIGV2ZW50cyIgLS0+IHYyNgp2MzcgLS0gInBvc3QgY29uc2Vuc3VzIHNpZ25hdHVyZXMiIC0tPiB2MzgKdjkgLS0gInBvc3NpYmxlIHBhcmVudCBsaXN0cyIgLS0+IHYxMAp2MTAgLS0gInBvc3NpYmxlIHBhcmVudHMiIC0tPiB2OAp2MjEgLS0gImV2ZW50cyB0byBoYXNoIiAtLT4gdjE0CnYyNCAtLSAiaGVhcnRiZWF0IiAtLT4gdjI1CnYzIC0tICJsaW5rZWQgZXZlbnRzIiAtLT4gdjIKdjMgLS0gImV2ZW50cyB0byBnb3NzaXAiIC0tPiB2NQp2MTkgLS0gIm5vbi1kZWR1cGxpY2F0ZWQgZXZlbnRzIiAtLT4gdjE3CnYzOCAtLSAiaXNzIG5vdGlmaWNhdGlvbnMiIC0tPiB2NDEKdjQxIC0tICJpc3Mgbm90aWZpY2F0aW9uIiAtLT4gdjM5CnY0MSAtLSAiSVNTIG5vdGlmaWNhdGlvbiIgLS0+IHY0MAp2NDEgLS0gIklTUyBub3RpZmljYXRpb24iIC0tPiB2NDIKdjI4IC0tICJldmVudHMgdG8gaGFzaCIgLS0+IHYxNAp2MjggLS0gImRvbmUgc3RyZWFtaW5nIHBjZXMiIC0tPiB2MjkKdjMyIC0tICJ1bmxpbmtlZCBldmVudHMiIC0tPiB2Mwp2MzIgLS0gImV2ZW50cyB0byB3cml0ZSIgLS0+IHYzMwp2MzMgLS0gImxhdGVzdCBkdXJhYmxlIHNlcXVlbmNlIG51bWJlciIgLS0+IHYzMQp2MTUgLS0gIm5vbi12YWxpZGF0ZWQgZXZlbnRzIiAtLT4gdjE5CnY1MCAtLSAicnVubmluZyBoYXNoIHVwZGF0ZSIgLS0+IHYwCnY1MCAtLSAicnVubmluZyBoYXNoIHVwZGF0ZSIgLS0+IHY0OQp2NDYgLS0gInN0YXRlIHNpZ25hdHVyZSB0cmFuc2FjdGlvbnMiIC0tPiB2MTEKY2xhc3NEZWYgczAgZmlsbDojY2NjLHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjAsdjEwLHYxMSx2MjEsdjI4LHYzMSx2MzIsdjM3LHYzOSx2NDAsdjQxLHY0Mix2NDQsdjUwIHMwCmNsYXNzRGVmIHMxIGZpbGw6IzlDRixzdHJva2U6IzAwMCxzdHJva2Utd2lkdGg6MnB4CmNsYXNzIHYxLHYxMyx2MTYsdjIwLHYyMyx2MjcsdjMwLHYzNSx2NDgsdjUxLHY3IHMxCmNsYXNzRGVmIHMyIGZpbGw6I2ZmOSxzdHJva2U6IzAwMCxzdHJva2Utd2lkdGg6MnB4CmNsYXNzIHYxNCx2MTUsdjE3LHYxOCx2MTksdjIsdjIyLHYyNCx2MjYsdjMsdjMzLHYzOCx2NDMsdjQ1LHY0Nix2NDcsdjQ5LHY1Mix2NTQsdjgsdjkgczIKY2xhc3NEZWYgczMgZmlsbDojZjg4LHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjEyLHYyNSx2MjksdjM0LHY0LHY1LHY1Myx2NTUsdjYgczMKY2xhc3NEZWYgczQgZmlsbDojQkVGLHN0cm9rZTojMDAwLHN0cm9rZS13aWR0aDoycHgKY2xhc3MgdjM2IHM0Cg==?bgColor=e8e8e8) (automatically generated from the source code).